frontend/packages/topology/src/components/quick-search/QuickSearchList.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import cx from 'classnames';
+import { history, removeQueryArgument } from '@console/internal/components/utils';
+import { CatalogItem } from '@console/plugin-sdk';
+import {
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListItemCells,
+  DataListCell,
+  Split,
+  SplitItem,
+  Label,
+  TextContent,
+  Text,
+  TextVariants,
+  Button,
+} from '@patternfly/react-core';
+import './QuickSearchList.scss';
+
+interface QuickSearchListProps {
+  listItems: CatalogItem[];
+  totalItems: number;
+  selectedItemId: string;
+  searchTerm: string;
+  namespace: string;
+  onSelectListItem: (itemId: string) => void;
+}
+
+const QuickSearchList: React.FC<QuickSearchListProps> = ({
+  listItems,
+  totalItems,
+  selectedItemId,
+  searchTerm,
+  namespace,
+  onSelectListItem,
+}) => {
+  const { t } = useTranslation();
+  const ref = React.useRef(null);
+
+  const openForm = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    const targetItemUID = ref.current?.props?.id;
+    const item = listItems.find((listItem) => listItem.uid === targetItemUID);
+    removeQueryArgument('catalogSearch');
+    history.push(item?.cta?.href);
+  };
+
+  const goToCatalogPage = React.useCallback(() => {
+    removeQueryArgument('catalogSearch');
+    history.push(`/catalog/ns/${namespace}?keyword=${searchTerm}`);
+  }, [namespace, searchTerm]);
+
+  return (
+    <DataList
+      className="odc-quick-search-list"
+      aria-label={t('topology~Quick Search List')}
+      selectedDataListItemId={selectedItemId}
+      onSelectDataListItem={(itemId) => itemId !== 'viewAll' && onSelectListItem(itemId)}
+      isCompact
+    >
+      {listItems.map((item) => (
+        <DataListItem
+          id={item.uid}
+          key={item.uid}
+          className={cx('odc-quick-search-list__item', {
+            'odc-quick-search-list__item--highlight': item.uid === selectedItemId,
+          })}
+          onDoubleClick={openForm}
+          ref={ref}
+        >
+          <DataListItemRow className="odc-quick-search-list__item-row">
+            <DataListItemCells
+              className="odc-quick-search-list__item-content"
+              dataListCells={[
+                <DataListCell isIcon key={`${item.uid}-icon`}>
+                  <img
+                    className="odc-quick-search-list__item-icon"
+                    src={item.icon?.url}
```

Comment:
So we have still a situation here where this can resolve to nothing and cause a blank. Perhaps we should check with @bgliwa01 if she prefers a blank/error image (depending on browser) or if she has a default to use.

Reply:
Icons aren't required but we have a fallback in catalog. Perhaps you need to use the same util and logic the catalog uses: `catalog-utils.tsx` `getIconProps`

Reply:
+1 to Christian...staying consistent with what the catalog has would be the ideal solution!

Reply:
Yes, @bgliwa01 this was resolved. :)